lib/rubycritic/generators/global_rating_calculator.rb

```diff
@@ -0,0 +1,15 @@
+module Rubycritic
+  module Generator
+    module GlobalRatingCalculator
+      def calculate_gpa(analysed_modules)
+        gpa_sum = 0
+        total_lines = 0
+        analysed_modules.each do |analysed_module|
+          gpa_sum += (analysed_module.rating.to_gpa) * analysed_module.lines
```

Comment:
I think converting the rating back to a GPA makes sense because it will put a cap on how much a really bad file can affect the total GPA. Since the rating can have an arbitrarily high cost it means that when averaged together a really bad F can skew the entire rating. I know at Spiceworks we have a model that is very very bad (over 4000 lines of code) so it would skew our GPA lower than if we made all F's the same.